### DISCUSSION TITLE
How to update OpenWrt?

### DISCUSSION THREAD

**Discussion by sgtjamez** [community-scripts/ProxmoxVE#908] [Q&A]
I have two instances of the ttneck script OpenWrt VM running. 
The older one is version:
OpenWrt 23.05.3 r23809-234f1a2efa / LuCI openwrt-23.05 branch git-24.346.66847-1bb28ba

The newest one is:
OpenWrt 23.05.5 r24106-10cc5fcd00 / LuCI openwrt-23.05 branch git-24.346.66847-1bb28ba 

Can someone tell me how to update OpenWrt? I don't think it's done using the Software portion of the interface, and typing 'update' from the VM console doesn't work. As new builds of the system come out, how am I able to upgrade?

Please and thank you.

**Comment by MickLesk** [comment]
Try this: 

**1. Update Package Information**
Update the package list to ensure the latest available updates:
```bash
opkg update
```

**2. Upgrade Installed Packages**
Upgrade all installed packages to the latest versions:

```bash
opkg upgrade
```
**3. System Upgrade (Optional)**
For a complete OpenWRT system upgrade, follow these steps:

a. Create a Backup
Backup your current system configuration:

```bash
sysupgrade -b /tmp/openwrt-backup.tar.gz
```
b. Download the New OpenWRT Image
Download the appropriate OpenWRT image from the OpenWRT Downloads page.

c. Upload the Image to the VM
Upload the new image to the VM:

```bash
scp openwrt-<version>-<target>.bin root@<VM_IP>:/tmp/
```
d. Perform the Upgrade
Start the upgrade with:

```bash
sysupgrade /tmp/openwrt-<version>-<target>.bin
```
The VM will automatically reboot after the upgrade.

**4. Verify and Restore (If Necessary)**
After upgrading, check the system. If issues arise, restore the backup:

```bash
sysupgrade -r /tmp/openwrt-backup.tar.gz
```

**Comment by sgtjamez** [comment]
Ok great. This is what I was looking into but needed a bit of a primer. For point 3.b., by ‘download’ does that mean wget? For the scp, I’m a little confused about where the source and destination are. But I think this is enough to get me going. I’m going to try it later today and I’ll come back with a post mortem. Thank you!Sent from my iPhoneOn Dec 19, 2024, at 7:55 AM, CanbiZ ***@***.***> wrote:﻿
Try this:
1. Update Package Information
Update the package list to ensure the latest available updates:
opkg update
2. Upgrade Installed Packages
Upgrade all installed packages to the latest versions:
opkg upgrade
3. System Upgrade (Optional)
For a complete OpenWRT system upgrade, follow these steps:
a. Create a Backup
Backup your current system configuration:
sysupgrade -b /tmp/openwrt-backup.tar.gz
b. Download the New OpenWRT Image
Download the appropriate OpenWRT image from the OpenWRT Downloads page.
c. Upload the Image to the VM
Upload the new image to the VM:
scp openwrt-<version>-<target>.bin root@<VM_IP>:/tmp/
d. Perform the Upgrade
Start the upgrade with:
sysupgrade /tmp/openwrt-<version>-<target>.bin
The VM will automatically reboot after the upgrade.
4. Verify and Restore (If Necessary)
After upgrading, check the system. If issues arise, restore the backup:
sysupgrade -r /tmp/openwrt-backup.tar.gz

**Reply by MickLesk** [reply]
C. Just Download the latest File and execute. This is howto is just for an ssh Client.
